File: appOcrTest/src/main/java/com/cartlc/ocrtest/data/PrefHelperBase.java
package com.cartlc.ocrtest.data;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by dug on 4/14/17.
 */

public class PrefHelperBase {

    protected SharedPreferences mPrefs;
    final protected Context mCtx;

    PrefHelperBase(Context ctx) {
        mCtx = ctx;
    }

    SharedPreferences getPrefs() {
        if (mPrefs == null) {
            mPrefs = mCtx.getSharedPreferences(getPrefFile(), 0);
        }
        return mPrefs;
    }

    String getPrefFile() {
        return mCtx.getPackageName() + "_preferences";
    }

    protected String getString(String key, String defaultValue) {
        return getPrefs().getString(key, defaultValue);
    }

    protected void setString(String key, String value) {
        SharedPreferences.Editor editor = getPrefs().edit();
        editor.putString(key, value);
        editor.commit();
    }
}

File: appMain/src/main/java/com/cartlc/tracker/model/pref/PrefHelperBase.kt
/**
 * Copyright 2018, FleetTLC. All rights reserved
 */
package com.cartlc.tracker.model.pref

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by dug on 4/14/17.
 */

open class PrefHelperBase constructor(protected val ctx: Context) {

    internal val prefs: SharedPreferences by lazy {
        ctx.getSharedPreferences(prefFile, 0)
    }

    private val prefFile: String
        get() = ctx.packageName + "_preferences"

    fun getString(key: String, defaultValue: String?): String? {
        return prefs.getString(key, defaultValue)
    }

    fun setString(key: String, value: String?) {
        val editor = prefs.edit()
        editor.putString(key, value)
        editor.apply()
    }

    fun getLong(key: String, defaultValue: Long): Long {
        return prefs.getLong(key, defaultValue)
    }

    fun setLong(key: String, value: Long) {
        val editor = prefs.edit()
        editor.putLong(key, value)
        editor.apply()
    }

    fun getInt(key: String, defaultValue: Int): Int {
        return prefs.getInt(key, defaultValue)
    }

    fun setInt(key: String, value: Int) {
        val editor = prefs.edit()
        editor.putInt(key, value)
        editor.apply()
    }
}
